### PR TITLE
Add java language

### PR DESCRIPTION
**Pull request by satyamsettyeswara**
Add the Random program in source directory

### CODE REVIEW

File: sources/Java.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
+import java.util.*;
+import java.io.*;
+import java.text.DecimalFormat;
+public class Java {
+
+	public static void main(String[] args) {
+		int range = Integer.parseInt(args[0]);
```
Comment (SpencerWoo) [inline sources/Java.java:7]: We can add error handling to gracefully fail here or even ask of inputs

File: sources/Java.java [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+		int trails = Integer.parseInt(args[1]);		
+		prng(range, trails);
+	}
+	static void prng(int numbers, int trails) {
+		int frequency[] = new int[numbers];
+		double probability[] = new double[numbers];
+		int start_time = (int) System.currentTimeMillis();
```
Comment (SpencerWoo) [inline sources/Java.java:14]: Why are we avoiding using the native `long`?  Generally, I'd like to remove all type casting.

File: sources/Java.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+		Random random = new Random();
+		for(int i=0;i<trails;i++) {
+			int r_number = random.nextInt(numbers);
+			frequency[r_number] +=1;
```
Comment (SpencerWoo) [inline sources/Java.java:18]: Source formatting to be consistent.  In some places we have spaces after `=` or `)`.

File: sources/Java.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+			if(frequency[i] == 0) {
+				output += i+" : 0\n";
+			}else {
+				probability[i] = frequency[i]/(double)trails;
+				DecimalFormat df = new DecimalFormat("#.######");				
+				output += i+" : "+df.format(probability[i])+"\n";
+			}
+		}
+		int duration = (int)System.currentTimeMillis() - start_time;
+		String filename = System.getProperty("user.dir")+"\\outputs\\"+"java_"+numbers+"_"+trails+".txt";
```
Comment (SpencerWoo) [inline sources/Java.java:31]: We should use `String.format` instead of concatenating strings with `+`.

File: sources/Java.java [hunk 5]
```diff
@@ -0,0 +1,41 @@
...
+		try {
+			BufferedWriter writer = new BufferedWriter(new FileWriter(filename));
```
Comment (SpencerWoo) [inline sources/Java.java:33]: I think we can write with `FileWriter` alone (without `BufferedWriter`)?  Let me know

**Review by SpencerWoo — CHANGES_REQUESTED** [review]
@satyamsettyeswara Amazing!  Thanks for this PR and following all the guidelines.  

Everything looks perfect except I'd like some refactoring on the java code.  All the functionality is there, just like it to be more consistent and clean.  I've left some comments.

I'll give it a few days in case you want to make those changes yourself (feel free to add commits to your current branch or rebase) -- otherwise I'm okay with merging it as-is and then I can make some SF/refactoring changes.

TLDR;
* Make requested changes
* If no changes are made, I'll merge as-is in a few days because it's a great PR and I can make the changes myself later

Let me know if there's any questions.  Otherwise thanks for contributing!  Hope to see another PR for another language =)

**Comment by SpencerWoo** [comment]
Can you also update the `sources/README.md` to move `Java` from Future to Completed language? 
 https://github.com/spencermwoo/randomness-in-programming-languages/pull/1/files#diff-a638ced7ac6ffb78c9770b5daef75b2e7749fed18298b0c2ec4df6a02002b05a

**Comment by satyamsettyeswara** [comment]
Added the requested changes. Tell me if you want any changes in the java code

File: sources/Java.java [hunk 6]
```diff
@@ -0,0 +1,51 @@
+import java.util.*;
+import java.io.*;
+import java.text.DecimalFormat;
+
+public class Java {
+
+	public static void main(String[] args) {
+		int range = 0;
+		int trails = 0;
```
Comment (spencermwoo) [inline sources/Java.java:9]: Let's rename this variable to `trials`

File: sources/Java.java [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+		for (long i = 0; i < trails; i++) {
+			int r_number = random.nextInt(numbers);
+			frequency[r_number] += 1;
+		}
+		String output = "";
+		for (int i = 0; i < numbers; i++) {
+			if (frequency[i] == 0) {
+				output += String.format("%d 0\n", i);
+			} else {
+				probability[i] = frequency[i] / (double) trails;
```
Comment (spencermwoo) [inline sources/Java.java:35]: `frequency[]` can be type `double` to avoid type-casting here

File: sources/Java.java [hunk 8]
```diff
@@ -0,0 +1,51 @@
...
+
+		prng(range, trails);
+	}
+
+	static void prng(int numbers, int trails) {
+		long frequency[] = new long[numbers];
+		double probability[] = new double[numbers];
+		long start_time = System.currentTimeMillis();
+		Random random = new Random();
+		for (long i = 0; i < trails; i++) {
```
Comment (spencermwoo) [inline sources/Java.java:26]: We aren't using `i` but it's best to keep it as `int`

**Review by spencermwoo — APPROVED** [review]
Thanks for making those changes.  This looks great.  I've left a few minor notes but I'll merge later in the week.

File: sources/Java.java [hunk 9]
```diff
@@ -0,0 +1,51 @@
...
+		long duration = System.currentTimeMillis() - start_time;
+		String filename = String.format("%s\\outputs\\java_%s_%s.txt", System.getProperty("user.dir"), numbers, trails);
+		try {
+			FileWriter fw = new FileWriter(filename);
+			fw.write(output);
+			fw.close();
+		} catch (Exception e) {
+			System.out.println(e);
+		}
+		System.out.println(String.format("--- %f seconds ---", (duration * 0.001)));
```
Comment (spencermwoo) [inline sources/Java.java:49]: We can also add newlines to separate the sections to help readability.

Typically that would be done with functions, but I enjoy one master function for this type of program.

**Comment by satyamsettyeswara** [comment]
Added all the requested changes

**Comment by SpencerWoo** [comment]
Thanks a ton @satyamsettyeswara - I've merged your pull request into master!
